swarm/runtime/safe_paths.py

```
@@ -46,3 +50,37 @@ def validate_path_component(component: str, name: str = "path component") -> str
+    Args:
+        ref: The string to validate.
+        name: Name of the variable for error messages.
+
+    Returns:
+        The validated ref (same as input).
+
+    Raises:
+        ValueError: If the ref is invalid.
+    """
+    if not ref:
+        raise ValueError(f"{name} cannot be empty")
```

Ruff (0.14.14)

[warning] Avoid specifying long messages outside the exception class

(TRY003)